# **World University Rankings 2026**

# Contents: 
1.	Importing libraries and data
2.	Exploring and Cleaning
3.	Checking for duplicates
4.	Renaming Columns
5.	Finding and Addressing Missing Values
6.	Exporting 

# 1. Importing libraries and data

In [1]:
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns
import scipy

In [2]:
path = r'C:\Users\lisad\OneDrive\Homework\Data Immersion\Achievement 6\University Rankings Project'

In [3]:
df = pd.read_excel(os.path.join(path, 'Data', 'Rankings2026.xlsx'), index_col = False)

# 2. Exploring

In [4]:
df.head()

2026 Rank Previous Rank                             Institution Name  \
0         1             1  Massachusetts Institute of Technology (MIT)   
1         2             2                      Imperial College London   
2         3             6                          Stanford University   
3         4             3                         University of Oxford   
4         5             4                           Harvard University   

          Country/Territory    Region Size Focus Research  \
0  United States of America  Americas    M    CO       VH   
1            United Kingdom    Europe    L    FO       VH   
2  United States of America  Americas    L    FC       VH   
3            United Kingdom    Europe    L    FC       VH   
4  United States of America  Americas    L    FC       VH   

                   Status  AR SCORE  ... ISR RANK  ISD SCORE ISD RANK  \
0  Private not for Profit     100.0  ...      153       92.3      130   
1                  Public      99.6  ...       35      100.0       22   
2  Private not for Profit     100.0  ...      261       76.1      230   
3                  Public     100.0  ...       80       98.7       67   
4  Private not for Profit     100.0  ...      217       60.6      335   

   IRN SCORE IRN RANK  EO SCORE EO RANK  SUS SCORE SUS RANK  Overall SCORE  
0       94.1       98     100.0       7       93.8       33            100  
1       97.5       28      95.9      68       98.3       7=           99.4  
2       96.5       49     100.0       2       95.4      19=           98.9  
3      100.0        1     100.0       3       77.9     198=           97.9  
4       99.4        8     100.0       1       77.8     201=           97.7  

[5 rows x 30 columns]

In [5]:
df.shape

(1501, 30)

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1501 entries, 0 to 1500
Data columns (total 30 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   2026 Rank          1501 non-null   object 
 1   Previous Rank      1389 non-null   object 
 2   Institution Name   1501 non-null   object 
 3   Country/Territory  1501 non-null   object 
 4   Region             1501 non-null   object 
 5   Size               1500 non-null   object 
 6   Focus              1501 non-null   object 
 7   Research           1500 non-null   object 
 8   Status             1454 non-null   object 
 9   AR SCORE           1501 non-null   float64
 10  AR RANK            1501 non-null   object 
 11  ER SCORE           1501 non-null   float64
 12  ER RANK            1501 non-null   object 
 13  FSR SCORE          1501 non-null   float64
 14  FSR RANK           1501 non-null   object 
 15  CPF SCORE          1501 non-null   float64
 16  CPF RANK           1501 

In [7]:
# 'Overall SCORE' would be better as a number (rather than an object)
# Replacing hyphens with nan
df['Overall SCORE'] = df['Overall SCORE'].replace('-', np.nan)

# Changing data type
df['Overall SCORE'] = df['Overall SCORE'].astype('float64')

C:\Users\lisad\AppData\Local\Temp\ipykernel_11160\2181057382.py:3: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Overall SCORE'] = df['Overall SCORE'].replace('-', np.nan)


In [8]:
# re-checking data types
df.dtypes

2026 Rank             object
Previous Rank         object
Institution Name      object
Country/Territory     object
Region                object
Size                  object
Focus                 object
Research              object
Status                object
AR SCORE             float64
AR RANK               object
ER SCORE             float64
ER RANK               object
FSR SCORE            float64
FSR RANK              object
CPF SCORE            float64
CPF RANK              object
IFR SCORE            float64
IFR RANK              object
ISR SCORE            float64
ISR RANK              object
ISD SCORE            float64
ISD RANK              object
IRN SCORE            float64
IRN RANK              object
EO SCORE             float64
EO RANK               object
SUS SCORE            float64
SUS RANK              object
Overall SCORE        float64
dtype: object

In [9]:
# Exploring descriptive statistics
df.describe()

AR SCORE     ER SCORE    FSR SCORE    CPF SCORE    IFR SCORE  \
count  1501.000000  1501.000000  1501.000000  1501.000000  1414.000000   
mean     25.785943    26.944237    33.950433    30.425516    36.305658   
std      24.500905    25.504494    28.440071    29.679882    35.252024   
min       1.000000     1.000000     1.000000     1.000000     1.000000   
25%       8.800000     8.500000    10.800000     6.000000     6.600000   
50%      16.000000    16.500000    23.500000    18.000000    20.100000   
75%      32.700000    37.500000    50.500000    49.700000    66.100000   
max     100.000000   100.000000   100.000000   100.000000   100.000000   

        ISR SCORE    ISD SCORE    IRN SCORE     EO SCORE    SUS SCORE  \
count  1464.00000  1464.000000  1499.000000  1501.000000  1477.000000   
mean     33.32541    34.526981    53.356905    29.989674    51.254367   
std      32.75066    31.108600    28.920632    29.197573    21.266331   
min       1.00000     1.000000     1.000000     1.000000     3.000000   
25%       5.90000     8.675000    27.500000     6.200000    35.700000   
50%      19.40000    21.700000    55.800000    17.900000    48.700000   
75%      56.70000    55.650000    78.500000    46.000000    66.500000   
max     100.00000   100.000000   100.000000   100.000000   100.000000   

       Overall SCORE  
count     703.000000  
mean       46.756046  
std        18.842125  
min        25.100000  
25%        30.900000  
50%        41.500000  
75%        58.900000  
max       100.000000

## 3. Checking for duplicates

In [10]:
df_dups = df[df.duplicated()]

In [11]:
df_dups.head()

Empty DataFrame
Columns: [2026 Rank, Previous Rank, Institution Name, Country/Territory, Region, Size, Focus, Research, Status, AR SCORE, AR RANK, ER SCORE, ER RANK, FSR SCORE, FSR RANK, CPF SCORE, CPF RANK, IFR SCORE, IFR RANK, ISR SCORE, ISR RANK, ISD SCORE, ISD RANK, IRN SCORE, IRN RANK, EO SCORE, EO RANK, SUS SCORE, SUS RANK, Overall SCORE]
Index: []

[0 rows x 30 columns]

## Verified no duplicates

## 4. Renaming Columns (for clarity)

In [12]:
df.rename(columns = {'AR SCORE' : 'Academic_Reputation_Score',
                     'AR RANK' : 'Academic_Reputation_Rank', 
                     'ER SCORE' : 'Employer_Reputation_Score', 
                     'ER RANK' : 'Employer_Reputation_Rank', 
                     'FSR SCORE' : 'Faculty_Student_Ratio_Score', 
                     'FSR RANK' : 'Faculty_Student_Ratio_Rank', 
                     'CPF SCORE' : 'Citations_Per_Faculty_Score', 
                     'CPF RANK' : 'Citations_Per_Faculty_Rank', 
                     'IFR SCORE' : 'International_Faculty_Ratio_Score', 
                     'IFR RANK' : 'International_Faculty_Ratio_Rank', 
                     'ISR SCORE' : 'International_Student_Ratio_Score', 
                     'ISR RANK' : 'International_Student_Ratio_Rank', 
                     'ISD SCORE' : 'International_Student_Diversity_Score', 
                     'ISD RANK' : 'International_Student_Diversity_Rank', 
                     'IRN SCORE' : 'International_Research_Network_Score', 
                     'IRN RANK' : 'International_Research_Network_Rank', 
                     'EO SCORE' : 'Employment_Outcomes_Score', 
                     'EO RANK' : 'Employment_Outcomes_Rank', 
                     'SUS SCORE' : 'Sustainability_Score',
                     'SUS RANK' : 'Sustainability_Rank'}, 
          inplace = True)


In [13]:
# Checking
df.columns

Index(['2026 Rank', 'Previous Rank', 'Institution Name', 'Country/Territory',
       'Region', 'Size', 'Focus', 'Research', 'Status',
       'Academic_Reputation_Score', 'Academic_Reputation_Rank',
       'Employer_Reputation_Score', 'Employer_Reputation_Rank',
       'Faculty_Student_Ratio_Score', 'Faculty_Student_Ratio_Rank',
       'Citations_Per_Faculty_Score', 'Citations_Per_Faculty_Rank',
       'International_Faculty_Ratio_Score', 'International_Faculty_Ratio_Rank',
       'International_Student_Ratio_Score', 'International_Student_Ratio_Rank',
       'International_Student_Diversity_Score',
       'International_Student_Diversity_Rank',
       'International_Research_Network_Score',
       'International_Research_Network_Rank', 'Employment_Outcomes_Score',
       'Employment_Outcomes_Rank', 'Sustainability_Score',
       'Sustainability_Rank', 'Overall SCORE'],
      dtype='object')

## 5. Finding and Addressing Missing Values

In [14]:
# Findng 
df.isnull().sum()

2026 Rank                                  0
Previous Rank                            112
Institution Name                           0
Country/Territory                          0
Region                                     0
Size                                       1
Focus                                      0
Research                                   1
Status                                    47
Academic_Reputation_Score                  0
Academic_Reputation_Rank                   0
Employer_Reputation_Score                  0
Employer_Reputation_Rank                   0
Faculty_Student_Ratio_Score                0
Faculty_Student_Ratio_Rank                 0
Citations_Per_Faculty_Score                0
Citations_Per_Faculty_Rank                 0
International_Faculty_Ratio_Score         87
International_Faculty_Ratio_Rank          87
International_Student_Ratio_Score         37
International_Student_Ratio_Rank          37
International_Student_Diversity_Score     37
Internatio

### Addressing 'Previous Rank' missing values

#### I pulled the 2025 data set and utilized an excel power query to confirm that the 112 institutions that do not have a 'previous rank' listed were not ranked on the 2025 list. Therefore, I'll impute 'NA' for the missing values. 

In [15]:
df['Previous Rank'].fillna('NA', inplace=True)

C:\Users\lisad\AppData\Local\Temp\ipykernel_11160\33836616.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Previous Rank'].fillna('NA', inplace=True)


In [16]:
# Checking
df['Previous Rank'].tail(25)

1476    1201-1400
1477    1201-1400
1478    1201-1400
1479    1201-1400
1480           NA
1481        1401+
1482    1201-1400
1483    1201-1400
1484        1401+
1485           NA
1486           NA
1487           NA
1488           NA
1489           NA
1490    1201-1400
1491           NA
1492        1401+
1493    1201-1400
1494    1201-1400
1495           NA
1496    1201-1400
1497    1201-1400
1498           NA
1499    1201-1400
1500    1201-1400
Name: Previous Rank, dtype: object

### Addressing 'Size' missing value

#### There is only one missing size value, Islamic Azad University. According to the QS website data, there are over a million students at Islamic Azad University, and QS defines anything more than 30,000 students to be size 'XL'. Therefore, I'm imputing 'XL' for the missing value. 

In [17]:
df['Size'].fillna('XL', inplace=True)

C:\Users\lisad\AppData\Local\Temp\ipykernel_11160\3214037365.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df['Size'].fillna('XL', inplace=True)


### Addressing 'Research' missing value

#### The only missing value for 'Research' is also Islmaic Azad University. I could not find any clarity around the intensity of research at that institution, so I'm going to just leave the missing value as is. 

### Addressing 'Status' missing values

In [18]:
# Identifying the missing entries
df_status_missing = df[df['Status'].isnull() == True]

In [19]:
df_status_missing

2026 Rank Previous Rank  \
81           82            NA   
261         262           236   
309         310            NA   
382         381           355   
487         487           380   
519         519           440   
529         530            NA   
572         571           489   
582         582           547   
784     781-790            NA   
799     791-800       691-700   
846     801-850            NA   
851     801-850       781-790   
852     851-900       771-780   
857     851-900            NA   
897     851-900       801-850   
919     901-950            NA   
922     901-950       901-950   
933     901-950       851-900   
934     901-950            NA   
985    951-1000      951-1000   
995    951-1000            NA   
1019  1001-1200     1201-1400   
1115  1001-1200      951-1000   
1136  1001-1200     1001-1200   
1138  1001-1200            NA   
1142  1001-1200      951-1000   
1154  1001-1200     1001-1200   
1157  1001-1200     1001-1200   
1159  1001-1200     1001-1200   
1162  1001-1200     1001-1200   
1181  1001-1200            NA   
1201  1201-1400            NA   
1240  1201-1400            NA   
1246  1201-1400     1201-1400   
1253  1201-1400     1201-1400   
1267  1201-1400            NA   
1300  1201-1400     1201-1400   
1312  1201-1400            NA   
1335  1201-1400     1201-1400   
1357  1201-1400     1201-1400   
1388  1201-1400     1201-1400   
1433      1401+     1201-1400   
1449      1401+            NA   
1452      1401+     1201-1400   
1463      1401+            NA   
1489      1401+            NA   

                                       Institution Name  \
81                                  Adelaide University   
261                              University of Aberdeen   
309              City St George’s, University of London   
382                            University of Luxembourg   
487        Technische Universität Bergakademie Freiberg   
519       Singapore University of Technology and Design   
529               The Education University of Hong Kong   
572                    Isfahan University of Technology   
582                               University of Iceland   
784                   Hong Kong Metropolitan University   
799         Norwegian University of Life Sciences (UMB)   
846                                 University of Namur   
851        Zurich University of Applied Sciences (ZHAW)   
852                              Addis Ababa University   
857                     Azerbaijan Technical University   
897                             University of Stavanger   
919                        Macao Polytechnic University   
922                       Osaka Metropolitan University   
933                           Universidad de Valladolid   
934                       Universidad Europea de Madrid   
985                     Universidad de Córdoba - España   
995                             University of Cyberjaya   
1019                                     CEU University   
1115                                Universidad de León   
1136                              Universitat de Lleida   
1138           Université de Bretagne Occidentale (UBO)   
1142                     Université Sorbonne Paris Nord   
1154                               University of Deusto   
1157                               University of Ibadan   
1159                             University of Ioannina   
1162                                University of Lagos   
1181                          University of the Algarve   
1201                     Ahmadu Bello University, Zaria   
1240                            Islamic Azad University   
1246                           Jahangirnagar University   
1253  Kwame Nkrumah University of Science and Techno...   
1267                        New Mexico State University   
1300              Technische Universität Kaiserslautern   
1312    UCAM Universidad Católica San Antonio de Murcia   
1335            Universidad Nacional de Ingeniería Peru   
13

#### Although there are 47 missing entries, a little research into each of the relevant universities (via their individual websites, and if needed Wikipe will allow me to find their statuses. After doing so, I'll create a dictionary in order to expedite the process of filling them in. 

In [20]:
# Fixing apostrophe mismatch before applying dictionary
df['Institution Name'] = df['Institution Name'].str.replace('’', "'", regex=False)

In [21]:
status_missing_dictionary = {'Adelaide University' : 'Public',
                             'University of Aberdeen' : 'Public', 
                             'City St George\'s, University of London' : 'Public', 
                             'University of Luxembourg' : 'Public',
                             'Technische Universität Bergakademie Freiberg' : 'Public', 
                             'Singapore University of Technology and Design' : 'Public', 
                             'The Education University of Hong Kong' : 'Public', 
                             'Isfahan University of Technology' : 'Public', 
                             'University of Iceland' : 'Public', 
                             'Hong Kong Metropolitan University' : ' Public', 
                             'Norwegian University of Life Sciences (UMB)' : 'Public', 
                             'University of Namur' : 'Public', 
                             'Zurich University of Applied Sciences (ZHAW)' : 'Public', 
                             'Addis Ababa University' : 'Public', 
                             'Azerbaijan Technical University' : 'Public', 
                             'University of Stavanger' : 'Public', 
                             'Macao Polytechnic University' : 'Public', 
                             'Osaka Metropolitan University' : 'Public', 
                             'Universidad de Valladolid' : 'Public', 
                             'Universidad Europea de Madrid' : 'Private for Profit', 
                             'Universidad de Córdoba - España' : 'Public', 
                             'University of Cyberjaya' : 'Private for Profit',
                             'CEU University' : 'Private not for Profit', 
                             'Universidad de León' : 'Public', 
                             'Universitat de Lleida' : 'Public', 
                             'Université de Bretagne Occidentale (UBO)' : 'Public', 
                             'Université Sorbonne Paris Nord' : 'Public', 
                             'University of Deusto' : 'Private not for Profit', 
                             'University of Ibadan' : 'Public', 
                             'University of Ioannina' : 'Public', 
                             'University of Lagos' : 'Public', 
                             'University of the Algarve' : 'Public',
                             'Ahmadu Bello University, Zaria' : 'Public',
                             'Islamic Azad University' : 'Private not for Profit', 
                             'Jahangirnagar University' : 'Public',
                             'Kwame Nkrumah University of Science and Technology' : 'Public', 
                             'New Mexico State University' : 'Public',
                             'Technische Universität Kaiserslautern' : 'Public', 
                             'UCAM Universidad Católica San Antonio de Murcia' : 'Private not for Profit',
                             'Universidad Nacional de Ingeniería Peru' : 'Public', 
                             'Université de Limoges' : 'Public',
                             'University of Rajshahi' : 'Public',
                             'Khulna University' : 'Public',
                             'Rajshahi University of Engineering and Technology' : 'Public',
                             'San Francisco State University' : 'Public',
                             'Tongmyong University' : 'Private not for Profit',
                             'University of Hawaii at Hilo' : 'Public'}



In [22]:
# Replacing missing values
df['Status'] = df['Status'].fillna(df['Institution Name'].map(status_missing_dictionary))

In [23]:
# Checking
df_status_check = df[df['Status'].isnull() == True]

In [24]:
df_status_check

Empty DataFrame
Columns: [2026 Rank, Previous Rank, Institution Name, Country/Territory, Region, Size, Focus, Research, Status, Academic_Reputation_Score, Academic_Reputation_Rank, Employer_Reputation_Score, Employer_Reputation_Rank, Faculty_Student_Ratio_Score, Faculty_Student_Ratio_Rank, Citations_Per_Faculty_Score, Citations_Per_Faculty_Rank, International_Faculty_Ratio_Score, International_Faculty_Ratio_Rank, International_Student_Ratio_Score, International_Student_Ratio_Rank, International_Student_Diversity_Score, International_Student_Diversity_Rank, International_Research_Network_Score, International_Research_Network_Rank, Employment_Outcomes_Score, Employment_Outcomes_Rank, Sustainability_Score, Sustainability_Rank, Overall SCORE]
Index: []

[0 rows x 30 columns]

### Addressing 'International_Faculty_Ratio_Score' and 'International_Faculty_Ratio_Rank' missing values 

#### These scores and rankings are calculated and provided by Qs. Since there are none provided for 87 institutions, I will trust that there is a logical reason for that and leave them null for now. 

### Addressing 'International_Student_Ratio_Score', 'International_Student_Ratio_Rank', 'International_Student_Diversity_Score', and 'International_Student_Diversity_Rank' missing values. 

#### Without extensive research (outside the scope of this project) there is no way to know why 37 institutions do not have values for these variables. Perhaps they have very few or no international students. Without a practical way to verify, I will leave them null for now. 

### Addressing 'International_Research_Network_Score' and 'International_Research_Network_Rank' missing values

In [25]:
# Identifying the institutions
df_missing_research_score = df[df['International_Research_Network_Score'].isnull() == True]

df_missing_research_score

2026 Rank Previous Rank                          Institution Name  \
929   901-950       741-750  Universidad Católica Andrés Bello - UCAB   
984  951-1000       781-790                   Universidad de Belgrano   

                      Country/Territory    Region Size Focus Research  \
929  Venezuela (Bolivarian Republic of)  Americas    M    FO       MD   
984                           Argentina  Americas    M    FO       MD   

                     Status  Academic_Reputation_Score  ...  \
929  Private not for Profit                       33.6  ...   
984  Private not for Profit                       11.2  ...   

    International_Student_Ratio_Rank  International_Student_Diversity_Score  \
929                             801+                                    9.2   
984                              503                                   29.2   

    International_Student_Diversity_Rank  \
929                                 801+   
984                                  624   

     International_Research_Network_Score International_Research_Network_Rank  \
929                                   NaN                                 NaN   
984                                   NaN                                 NaN   

     Employment_Outcomes_Score Employment_Outcomes_Rank  Sustainability_Score  \
929                       22.6                      677                   NaN   
984                       44.3                      397                   NaN   

    Sustainability_Rank  Overall SCORE  
929                 NaN            NaN  
984                 NaN            NaN  

[2 rows x 30 columns]

In [26]:
df_missing_research_rank = df[df['International_Research_Network_Rank'].isnull() == True]

df_missing_research_rank

2026 Rank Previous Rank                          Institution Name  \
929   901-950       741-750  Universidad Católica Andrés Bello - UCAB   
984  951-1000       781-790                   Universidad de Belgrano   

                      Country/Territory    Region Size Focus Research  \
929  Venezuela (Bolivarian Republic of)  Americas    M    FO       MD   
984                           Argentina  Americas    M    FO       MD   

                     Status  Academic_Reputation_Score  ...  \
929  Private not for Profit                       33.6  ...   
984  Private not for Profit                       11.2  ...   

    International_Student_Ratio_Rank  International_Student_Diversity_Score  \
929                             801+                                    9.2   
984                              503                                   29.2   

    International_Student_Diversity_Rank  \
929                                 801+   
984                                  624   

     International_Research_Network_Score International_Research_Network_Rank  \
929                                   NaN                                 NaN   
984                                   NaN                                 NaN   

     Employment_Outcomes_Score Employment_Outcomes_Rank  Sustainability_Score  \
929                       22.6                      677                   NaN   
984                       44.3                      397                   NaN   

    Sustainability_Rank  Overall SCORE  
929                 NaN            NaN  
984                 NaN            NaN  

[2 rows x 30 columns]

#### According to the QS website's Methodology section, the International Research Network is a measure of an institution's success in creating and sustaining research partnerships with institutions in other locations. It is very likely that these 2 institutions have minimal if any sustained research partnerships. It is outside the scope of this project to look into that further. Therefore, I will leave these values as null as well. 

### Addressing 'Sustainablity_Score' and 'Sustainablity_Rank' missing values

#### According to the QS website's Methodology secgtion, not all institutions are eligible and included in sustainability measures. Therefore, I will again, leave the null values. 

### Addressing 'Overall Score' missing values

#### Starting with a 2026 Rank of 701, institutions are grouped into bands, at which point an overall score is no longer assigned. 

## Addressing Missing Values Completed

# Exporting cleaned dataset

In [27]:
df.to_excel(os.path.join(path, 'Data', 'Rankings2026_cleaned.xlsx'))